test/functional/api/ui_spec.lua

```diff
@@ -0,0 +1,39 @@
+local helpers = require('test.functional.helpers')(after_each)
+local Screen = require('test.functional.ui.screen')
+local clear = helpers.clear
+local eq = helpers.eq
+local eval = helpers.eval
+local request = helpers.request
+
+describe('nvim_ui_attach()', function()
+  before_each(function()
+    clear()
+  end)
+  it('handles very large width/height #2180', function()
+    local screen = Screen.new(999, 999)
+    screen:attach()
+    eq(999, eval('&lines'))
+    eq(999, eval('&columns'))
+  end)
+  it('invalid option returns error', function()
+    local screen = Screen.new()
+    local status, rv = pcall(function() screen:attach({foo={'foo'}}) end)
+    eq(false, status)
+    eq('No such UI option', rv:match("No such .*"))
+  end)
+  it('validates channel arg', function()
+    assert.has_error(function() request('nvim_ui_try_resize', 40, 10) end,
```

Comment:
Normally we use pcall for this, this doesn't gain more than it loses IMHO.
But I guess we could add wrapper such that we could write `eq('UI not attached to channel: 1', get_err(meths.ui_try_resize, 40, 10))`

Reply:
What did this lose?

Reply:
I mean in length and readability, I don't think we should start using a style where a lambda is required to assert every error return of an ordinary API call.  I just realised `eq({false, errmsg}, {pcall(method, args)})` should work, though it looks a bit weird if not used to lua arglist semantics...

Reply:
@bfredl I like that idea. But `pcall` error message also contains the source code info:

    'test/functional/helpers.lua:94: UI not attached to channel: 1'

Is there an easy way to avoid that?

Reply:
luassert uses the following pattern:
```
    -- remove 'path/to/file:line: ' from string
    err_actual = err_actual:gsub('^.-:%d+: ', '', 1)
```